service/src/integrationTest/java/uk/gov/hmcts/reform/fpl/controllers/ApplicantMidEventControllerTest.java

```diff
@@ -0,0 +1,71 @@
+package uk.gov.hmcts.reform.fpl.controllers;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+import org.assertj.core.api.Assertions;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.OverrideAutoConfiguration;
+import org.springframework.boot.test.autoconfigure.web.servlet.WebMvcTest;
+import org.springframework.http.MediaType;
+import org.springframework.test.context.ActiveProfiles;
+import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.test.web.servlet.MvcResult;
+import uk.gov.hmcts.reform.ccd.client.model.AboutToStartOrSubmitCallbackResponse;
+import uk.gov.hmcts.reform.ccd.client.model.CallbackRequest;
+import uk.gov.hmcts.reform.ccd.client.model.CaseDetails;
+import uk.gov.hmcts.reform.fpl.model.Applicant;
+import uk.gov.hmcts.reform.fpl.model.ApplicantParty;
+
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.post;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+
+
+@ActiveProfiles("integration-test")
+@WebMvcTest(ApplicantController.class)
+@OverrideAutoConfiguration(enabled = true)
+public class ApplicantMidEventControllerTest {
+    private static final String AUTH_TOKEN = "Bearer token";
+    private static final String USER_ID = "1";
+    private static final ObjectMapper MAPPER = new ObjectMapper();
+    private static final String ERROR_MESSAGE = "Payment by account (PBA) number must include 7 numbers";
+
+    @Autowired
+    private MockMvc mockMvc;
+
+    @Test
+    void shouldReturnErrorsWhenThereIsNewApplicantAndNoPbaNumber() throws Exception {
```

Comment:
Is having one tests for new structure enough while old structure had more than that? Please cover same test cases or more as old case structure had covered.

Reply:
👍

Reply:
is this the right place for these tests? Integration tests can be slow, I would have this covered in unit tests. (I have no got there yet in your code review to see what there is). From memory, the pba number bits are all in a separate service